vllm/model_executor/models/whisper.py

```diff
@@ -520,24 +572,15 @@ def __init__(
                 sinusoids(*self.embed_positions.weight.shape)
             )
 
-    def forward(self, input_features: torch.Tensor | list[torch.Tensor]):
-        hidden_states = []
-        input_is_batched = False
-        for features in input_features:
-            embeds = nn.functional.gelu(self.conv1(features))
-            embeds = nn.functional.gelu(self.conv2(embeds))
-            embeds = embeds.transpose(-1, -2)
-            embeds = (embeds + self.embed_positions.weight[: embeds.size(-2), :]).to(
-                embeds.dtype
-            )
-            hidden_states.append(embeds)
-            input_is_batched = embeds.ndim > 2
+    def forward(self, input_features: torch.Tensor):
+        # print("input_features", input_features.shape, "\n\n", flush=True)
+        embeds = nn.functional.gelu(self.conv1(input_features))
+        embeds = nn.functional.gelu(self.conv2(embeds))
+        embeds = embeds.transpose(-1, -2)
```

Comment:
signature here is changed, but it allows for a small win by ditching the slow for loop in favor of proper tensor ops